app/views/govuk_publishing_components/components/_list.html.erb

```diff
@@ -0,0 +1,26 @@
+<%
+  id ||= nil
+  extra_spacing ||= nil
+  list_style ||= nil
+  list_type ||= "unordered"
+  visible_counters ||= nil
+  items ||= []
```

Comment:
I haven't fully formulated this thought yet but it's around the idea that you have to pass a chunk of markup for each list item. I can't think of another way of doing it, because anything else would restrict what you could display and you don't want that (say, if the list item is text with two links in, or even some more complex formatting).

I think my concern is that for some uses of this component it would produce a situation where either the view gets quite cluttered or we start assembling markup outside of the view and injecting it. I don't know. Let's have a conversation about this. Thoughts? Is this a concern?

Reply:
On that subject, could you put an example into the YML of a list that contains links?

Reply:
Sure - I'll add an example. It'll be the same as the example with an `&rsquo;` in it and it's always good to be clearer.

I think adding markup into each list item is okay, as other options are much too restrictive. We allow passing in markup into other components - such as the accordion - and having elements inside a `li` is allowed by the HTML spec. Every string is put through `sanitize()`, which should help stop it breaking if some erroneous markup is used.